FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_DNS.c

```diff
@@ -775,74 +800,91 @@ static const DNSMessage_t xDefaultPartDNSHeader =
 									   size_t uxDestLen )
 	{
 	size_t uxNameLen = 0;
-	BaseType_t xCount;
+	size_t uxLocalSourceLen = uxSourceLen;
```

Comment:
Why these redundant stack variables?

Reply:
We cannot directly modify the parameters being passed to the function. While totally leagal in C, MISRA advises against it in rule 17.8.
These redundant stack variables will be eventually removed by compiler with proper optimization level.